cmd/apiserver/main.go

```diff
@@ -31,9 +26,8 @@ type options struct {
 	port   int
 }
 
-func gatherOptions() options {
+func Bind(fs *flag.FlagSet) options {
```

Comment:
you shouldn't rename this at all.

Reply:
I must have missunderstood this:
Also please change the name of this func to Bind(fs *flag.FlagSet) and use it to bind the flags that are related to the simulator to gatherOptions in main.

Reply:
Yea. Let me explain this further. In `simulator` struct you should have a method `Bind(fs *flag.FlagSet)` that will take care only the flags that are related with the simulator. Then in `gatherOptions` you can do something like `o.simulator.Bind(fs)`